VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -25,17 +25,924 @@ if ($PSEdition -eq 'Desktop') {
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $pass,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $domain,
+        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String] $cluster, 
+        [Parameter (Mandatory = $true, ParameterSetName = "enable")] [ValidateNotNullOrEmpty()] [Switch] $enable,
+        [Parameter (Mandatory = $true, ParameterSetName = "disable")] [ValidateNotNullOrEmpty()] [Switch] $disable
+
+    )
+    Try {
+        $vcfVcenterDetails = Get-vCenterServerConnection -server $server -user $user -pass $pass -domain $domain
+        if (Get-Cluster | Where-Object { $_.Name -eq $cluster }) {
+            $esxiHosts = Get-Cluster $cluster | Get-VMHost | Sort-Object -Property Name
+            if (!$esxiHosts) { Write-Warning "No ESXi hosts found within $cluster cluster." }
+        }
+        else {
+            Write-Error "Unable to locate Cluster $cluster in $($vcfVcenterDetails.fqdn) vCenter Server: PRE_VALIDATION_FAILED" -ErrorAction Stop
+        }
+
+        if ($PSBoundParameters.ContainsKey("enable")) {
+
+            Write-Host -ForegroundColor Yellow "Enabling Lockdown Mode on all hosts in the $ClusterName cluster."
+            ForEach ($esxiHost in $esxiHosts) {
+                $currentMode = (Get-VMHost -name $esxiHost).ExtensionData.Config.LockdownMode
+                if ($currentMode -eq "lockdownDisabled") {
+                    ($esxiHost | Get-View).EnterLockdownMode()
+                    Write-Host "Changing $esxiHost mode from $currentMode to lockdownNormal"
+                    $newMode = (Get-VMHost -name $esxiHost).ExtensionData.Config.LockdownMode
+                    if ($currentMode -eq $newMode) {
+                        Write-Error "Unable to change $esxiHost mode from $currentMode to lockdownNormal. Currently it is in $newMode" -ErrorAction Stop
+                    }
+                }
+                else {
+                    Write-Host "$esxiHost is in already in lockdownNormal mode"
+                }
+            }
+        } 
+        
+        if ($PSBoundParameters.ContainsKey("disable")) {
+            Write-Host -ForegroundColor Yellow "Disabling Lockdown Mode on all hosts in the $ClusterName cluster."
+            ForEach ($esxiHost in $esxiHosts) {
+                $currentMode = (Get-VMHost -name $esxiHost).ExtensionData.Config.LockdownMode
+                if ($currentMode -ne "lockdownDisabled") {
+                    ($esxiHost | Get-View).ExitLockdownMode()
+                    Write-Host "Changing $esxiHost mode from $currentMode to lockdownDisabled"
+                    $newMode = (Get-VMHost -name $esxiHost).ExtensionData.Config.LockdownMode
+                    if ($currentMode -eq $newMode) {
+                        Write-Error "Unable to change $esxiHost mode from $currentMode to lockdownDisabled. Currently it is in $newMode" -ErrorAction Stop
```

Comment:
```suggestion
                        Write-Error "Unable to change lockdown mode for ESXi host $esxiHost from $currentLockdownMode to lockdownNormal. Lockdown mode is set to $newLockdownMode." -ErrorAction Stop
```

Reply:
it should be lockdownDisabled instead of lockdownNormal. Made this change.